igcollect/linux_pressure.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+"""igcollect - PSI - Pressure Stall Information
+
+Copyright (c) 2020 InnoGames GmbH
+"""
+
+import os
+import time
+
+from argparse import ArgumentParser
+
+
+def parse_args():
+    parser = ArgumentParser(prog='linux_pressure.py')
+    parser.add_argument('--prefix', default='pressure')
+    return parser.parse_args()
+
+
+def main():
+    if os.path.isdir('/proc/pressure'):
+        args = parse_args()
+        for pressure in get_pressure_stats():
+            print('{}.{}'.format(args.prefix, pressure))
+
+
+def parse_split_file(filename):
+    """Utility to read and parse a space delimited file"""
+    with open(filename, 'r') as fd:
+        return [line.strip().split(' ') for line in fd]
```

Comment:
How long can that file be? If too long, you might want to consider returning a generator instead of a list by replacing the brackets to braces:
```suggestion
        return (line.strip().split(' ') for line in fd)
```

Reply:
Size in bytes:
> 56 /proc/pressure/cpu
> 110 /proc/pressure/io
> 110 /proc/pressure/memory
> 276 total

I consider this rather short.

Reply:
I tried your suggestion:
> ` return (line.split() for line in fd)`
and it breaks on:
>    File "linux_pressure.py", line 29, in <genexpr>
>     return (line.split() for line in fd)
> ValueError: I/O operation on closed file

will keep the code inside square brakets

Reply:
Yeah, 276 bytes is not interesting at all. You can ignore that then. It's more interesting if it's about hundreds of MB, then to not overload the memory with it.
The error makes sense, the file descriptor `fd` is in the generator, if we would need the generator, then we would need to change the process there.
But like I said, 276 bytes are totally ignorable.